moveit_core/trajectory_processing/test/test_time_optimal_trajectory_generation.cpp

```diff
@@ -538,6 +538,29 @@ TEST(time_optimal_trajectory_generation, testSingleDofDiscontinuity)
   }
 }
 
+TEST(time_optimal_trajectory_generation, testRelevantZeroMaxAccelerationsInvalidateTrajectory)
+{
+  EXPECT_FALSE(Trajectory(Path(std::list<Eigen::VectorXd>{ Eigen::Vector2d(0, 0), Eigen::Vector2d(1, 1) }),
+                          Eigen::Vector2d(1, 1), Eigen::Vector2d(0, 1))
+                   .isValid());
+  EXPECT_FALSE(Trajectory(Path(std::list<Eigen::VectorXd>{ Eigen::Vector2d(0, 0), Eigen::Vector2d(1, 1) }),
+                          Eigen::Vector2d(1, 1), Eigen::Vector2d(1, 0))
+                   .isValid());
+  EXPECT_FALSE(Trajectory(Path(std::list<Eigen::VectorXd>{ Eigen::Vector2d(0, 0), Eigen::Vector2d(1, 1) }),
+                          Eigen::Vector2d(1, 1), Eigen::Vector2d(0, 0))
+                   .isValid());
+}
+
+TEST(time_optimal_trajectory_generation, testIrrelevantZeroMaxAccelerationsDontInvalidateTrajectory)
+{
+  EXPECT_TRUE(Trajectory(Path(std::list<Eigen::VectorXd>{ Eigen::Vector2d(0, 0), Eigen::Vector2d(0, 1) }),
+                         Eigen::Vector2d(1, 1), Eigen::Vector2d(0, 1))
+                  .isValid());
+  EXPECT_TRUE(Trajectory(Path(std::list<Eigen::VectorXd>{ Eigen::Vector2d(0, 0), Eigen::Vector2d(1, 0) }),
+                         Eigen::Vector2d(1, 1), Eigen::Vector2d(1, 0))
+                  .isValid());
+}
```

Comment:
Great that you added unit tests -- for my reference, where in this test are we specifying that max acceleration / max vel is zero?

Reply:
Third parameter of Trajectory() is the maximum acceleration (see that tests set different components of the acceleration to 0, or all of them). Zero max velocities did not make the loop to spin indefinitely.

Reply:
I'm going to add parameter annotations to the calls in the tests to make it clearer.

Reply:
Added parameter annotations.